### ISSUE TITLE
WarnServ needs to be rewritten to use Postgres

### ISSUE THREAD

**Issue by shymega** [vktec-irc/warnserv#14]
@vktec - you discussed on IRC that you were going to rewrite WarnServ
to use Postgres to manage warnings due to its growing complexity,
IIRC, using your custom-built CHICKEN DB ORM.

I have opened this issue to remind us all :-)

Cc: @vktec

**Comment by vktec** [comment]
Well, I've kinda scrapped that ORM (too much work for too little gain) but SQL isn't hard, so I'll just be using that.

**Comment by shymega** [comment]
Hello,

Samadi van Koten <notifications@github.com> writes:

> Well, I've kinda scrapped that ORM (too much work for too little
> gain)

Right. Fair enough.

> but SQL isn't hard, so I'll just be using that.

Bear in mind- it's not just SQL. You need to understand the concepts
of database design as well, such as referential integrity,
normalisation to ensure a healthy and stable database.

I got that from my BTEC IT Course from College- Unit 18, Database
design. Interesting stuff, and worth bearing in mind.

-- 
Best regards,
Dom Rodriguez

**Comment by vktec** [comment]
Yes, that's also important but, in my opinion, ORMs can be both a blessing and a curse in that way: they manage it all for you, but they abstract it away so you don't know what's going on.

It's a similar thing to what a certain large tech company who makes very slick looking but rather featureless products does.

**Comment by shymega** [comment]
Hello,

Samadi van Koten <notifications@github.com> writes:

> Yes, that's also important but, in my opinion, ORMs can be both a
> blessing and a curse in that way: they manage it all for you, but
> they abstract it away so you don't know what's going on.

Yesh. Stupid abstractions.

> It's a similar thing to what a certain large tech company who makes
> very slick looking but rather featureless products does.

Hmmmmm... I wonder *who* that could be? :-P

-- 
Best regards,
Dom Rodriguez

**Comment by vktec** [comment]
Hehe
